lib/capistrano/ext/multistage.rb

```diff
@@ -22,7 +22,7 @@
   on :load do
     if stages.include?(ARGV.first)
       # Execute the specified stage so that recipes required in stage can contribute to task list
-      find_and_execute_task(ARGV.first) if ARGV.any?{ |option| option =~ /-T|--tasks|-e|--explain/ }
+      find_and_execute_task(ARGV.first) unless (ARGV & %w(-T --tasks -e --explain)).empty?
```

Comment:
I made a conscious decision to intersect the arrays instead of using `any?` + `include?`.  Iterating is only faster if the arrays are very small, or if there's a match near the start.  These are seldom used arguments, so more often than not it would need to traverse both arrays to return no match.

In hindsight, I should have optimized for readability instead of performance.

I'll create a new PR off the tip of the `legacy-v2` branch and submit it upstream.  This patch was branched off the version tag that we're using.